pwndbg/dbg/lldb/repl/__init__.py

```diff
@@ -129,7 +130,25 @@ def created(self):
         self.dbg._trigger_event(EventType.START)
 
     @override
-    def suspended(self):
+    def suspended(self, event: lldb.SBEvent):
+        # The event might have originated from a different source than the user
+        # currently has selected. Move focus to the where the event happened.
+        #
+        # state-changed events have no thread associated with them, and so
+        # SBThread::GetThreadFromEvent does not work. Interrogate each thread in
+        # the process and look for the most interesting one.
+        proc = lldb.SBProcess.GetProcessFromEvent(event)
+        for thread in proc.threads:
+            # Currently the one considered most interesting is simply the first
+            # that has any reason at all to be stopped.
+            if thread.stop_reason == lldb.eStopReasonNone:
+                continue
+
+            if proc.GetSelectedThread().idx != thread.idx:
+                print(message.notice(f"[Switched to Thread {thread.id}]"))
+                assert proc.SetSelectedThread(thread)
```

Comment:
If you prefer this be made into a warning, just let me know.

Reply:
Pls make it a warning instead; its better to not crash if possible

Reply:
> Pls make it a meaningful warning instead; its better to not crash if possible

I get that, but, are you sure? This failing indicates that either LLDB is giving us nonsense values or that Pwndbg is in some really broken state. I feel like maybe we want it to go down loudly in that case.